app/controllers/posts_controller.rb

```diff
@@ -237,6 +238,20 @@ def destroy_many
     render nothing: true
   end
 
+  def merge_posts
+    params.require(:post_ids)
+
+    posts = Post.where(id: post_ids_including_replies).order(:id)
+    raise Discourse::InvalidParameters.new(:post_ids) if posts.blank?
```

Comment:
Sorry for the delay in getting back to you :(

You can do something like `posts.pluck(:id) == params[:post_ids]`.